src/python/pants/build_graph/files.py

```diff
@@ -6,19 +6,28 @@
 
 
 class Files(Target):
-  """A collection of loose files."""
+  """A collection of loose files.
+
+  Exactly one of `sources` or `urls` should be specified.
```

Comment:
Responding to your questions, I think I prefer allowing `sources=` to accept URLs, and on any target. 

Whenever I see comments restricting possible combinations of arguments I think "there's a design problem here".